go/oasis-node/cmd/debug/txsource/workload/registration.go

```diff
@@ -0,0 +1,343 @@
+
+func getRuntime(entityID signature.PublicKey) *registry.Runtime {
+	rt := &registry.Runtime{
+		ID:       ns,
+		EntityID: entityID,
+		Kind:     registry.KindCompute,
+		Executor: registry.ExecutorParameters{
+			GroupSize:    1,
+			RoundTimeout: 1 * time.Second,
+		},
+		Merge: registry.MergeParameters{
+			GroupSize:    1,
+			RoundTimeout: 1 * time.Second,
+		},
+		TxnScheduler: registry.TxnSchedulerParameters{
+			GroupSize:         1,
+			Algorithm:         "batching",
+			BatchFlushTimeout: 1 * time.Second,
+			MaxBatchSize:      1,
+			MaxBatchSizeBytes: 1,
+		},
+		Storage: registry.StorageParameters{
+			GroupSize:               1,
+			MaxApplyWriteLogEntries: 100_000,
+			MaxApplyOps:             2,
+			MaxMergeRoots:           8,
+			MaxMergeOps:             2,
+		},
+		AdmissionPolicy: registry.RuntimeAdmissionPolicy{
+			AnyNode: &registry.AnyNodeRuntimeAdmissionPolicy{},
+		},
+	}
+	rt.Genesis.StateRoot.Empty()
+	return rt
+}
+
+func getNodeDesc(rng *rand.Rand, nodeIdentity *identity.Identity, entityID signature.PublicKey) *node.Node {
+	nodeAddr := node.Address{
+		TCPAddr: net.TCPAddr{
+			IP:   net.IPv4(127, 0, 0, 1),
+			Port: 12345,
+			Zone: "",
+		},
+	}
+
+	// NOTE: we shouldn't be registering validators, as that would lead to
+	// consensus stopping as the registered validators wouldn't actually
+	// exist.
+	availableRoles := []node.RolesMask{
+		node.RoleStorageWorker,
+		node.RoleComputeWorker,
+		node.RoleStorageWorker | node.RoleComputeWorker,
+	}
+
+	nodeDesc := node.Node{
+		ID:         nodeIdentity.NodeSigner.Public(),
+		EntityID:   entityID,
+		Expiration: 0,
+		Roles:      availableRoles[rng.Intn(len(availableRoles))],
+		Committee: node.CommitteeInfo{
+			Certificate: nodeIdentity.TLSCertificate.Certificate[0],
+			Addresses: []node.CommitteeAddress{
+				{
+					Certificate: nodeIdentity.TLSCertificate.Certificate[0],
+					Address:     nodeAddr,
+				},
+			},
+		},
+		P2P: node.P2PInfo{
+			ID: nodeIdentity.P2PSigner.Public(),
+			Addresses: []node.Address{
+				nodeAddr,
+			},
+		},
+		Consensus: node.ConsensusInfo{
+			ID: nodeIdentity.ConsensusSigner.Public(),
+			Addresses: []node.ConsensusAddress{
+				{
+					ID:      nodeIdentity.ConsensusSigner.Public(),
+					Address: nodeAddr,
+				},
+			},
+		},
+		Runtimes: []*node.Runtime{
+			{
+				ID: ns,
+			},
+		},
+	}
+	return &nodeDesc
+}
+
+func signNode(identity *identity.Identity, nodeDesc *node.Node) (*node.MultiSignedNode, error) {
+	nodeSigners := []signature.Signer{
+		identity.NodeSigner,
+		identity.P2PSigner,
+		identity.ConsensusSigner,
+		identity.TLSSigner,
+	}
+
+	sigNode, err := node.MultiSignNode(nodeSigners, registry.RegisterNodeSignatureContext, nodeDesc)
+	if err != nil {
+		registryLogger.Error("failed to sign node descriptor",
+			"err", err,
+		)
+		return nil, err
+	}
+
+	return sigNode, nil
+}
+
+func getAccountNonce(ctx context.Context, conn *grpc.ClientConn, pubKey signature.PublicKey) (uint64, error) {
```

Comment:
Maybe we should expose `TransactionAuthHandler` in `consensus.ClientBackend` instead (it has a `GetSignerNonce` method)?

Reply:
yeah maybe? Probably would actually need to expose the `GetSignerNonce` method directly so it can work with the grpc backend? Could just embed the `TransactionAuthHandler` in `ClientBackend`?

Reply:
done